patient-care/webapp/lib/methods.js

```diff
@@ -208,136 +208,267 @@ Meteor.methods({
-      let nodes = {};
+    let user = MedBook.ensureUser(Meteor.userId());
 
-      _.each(samplesByDataSetId, (sampleLabels, data_set_id) => {
-        // data set security
-        let dataSet = DataSets.findOne(data_set_id);
-        user.ensureAccess(dataSet);
+    let dataSet = DataSets.findOne(args.data_set_id);
+    user.ensureAccess(dataSet);
 
-        // initialize nodes[sampleLabels] to put gene data there
-        _.each(sampleLabels, (label) => { nodes[label] = {}; });
+    args.data_set_name = dataSet.name;
 
-        // load the data for this data set
-        GeneExpression.find({ data_set_id }).forEach((doc) => {
-          _.each(sampleLabels, (sample_label) => {
-            let sampleIndex = dataSet.gene_expression_index[sample_label];
-            let expValue = doc.rsem_quan_log2[sampleIndex];
+    let sampleLabels = args.sample_labels;
+    delete args.sample_labels;
 
-            nodes[sample_label][doc.gene_label] = expValue;
-          });
-        });
-      });
-      console.log("done loading data");
-
-      // do this to allow non-SSL connections (I think)
-      process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
-
-      // do the API call
-      apiResponse = HTTP.call("POST",
-          "https://tumormap.ucsc.edu:8112/query/overlayNodes", {
-        data: {
-          map: "CKCC/v1",
-          layout: "mRNA",
-          nodes
-        }
+    return _.map(sampleLabels, (sample_label) => {
+      // will override previous values
+      args.sample_label = sample_label;
+
+      return Jobs.insert({
+        name: "RunSingleSampleTopGenes",
+        status: "waiting",
+        user_id: user._id,
+        collaborations: [ user.personalCollaboration() ],
+        args
       });
+    });
+  },
+  createLimma(args) {
+    check(args, new SimpleSchema({
+      // No allowed values for value_type becuase we check if the value_types
+      // of the sample groups match this, and those have allowedValues on
+      // insert.
+      value_type: { type: String },
 
-      if (apiResponse.statusCode === 200) {
-        Jobs.update(jobId, {
-          $set: {
-            status: "done",
-            output: {
-              // TODO: should be `bookmark`
-              bookmark_url: apiResponse.data.bookmarks[0],
-            }
-          }
-        });
-      } else {
-        Jobs.update(jobId, { $set: { status: "error" } });
+      experimental_sample_group_id: { type: String },
+      reference_sample_group_id: { type: String },
+      top_genes_count: { type: Number, min: 1 },
+    }));
+
+    // ensure access to all of the things
+    let user = MedBook.ensureUser(Meteor.userId());
+
+    let experimental = SampleGroups.findOne(args.experimental_sample_group_id);
+    let reference = SampleGroups.findOne(args.reference_sample_group_id);
+    user.ensureAccess(experimental);
+    user.ensureAccess(reference);
+
+    // TODO: make sure the groups don't share samples from the same data set
```

Comment:
Fine to not do this right now, since you added the helpful error message for when limma fails due to this.
And also to be specific it's "make sure the groups don't share the same sample with expression" ; non-overlapping samples from the same data set is totally fine.